src/renderer/components/pages/Config/pages/LaunchConfig/LaunchConfig.tsx

```diff
@@ -0,0 +1,164 @@
+import { LoadingOverlay } from '@/renderer/components/common/LoadingOverlay'
+import { useActor } from '@xstate/react'
+import { rosService } from '@/renderer/state/ros'
+
+interface LaunchMsg {
+  fileName: string
+  message: string
+  isLaunched: boolean
+}
+
+interface LaunchedFiles {
+  packages: string[]
+  fileNames: string[]
+}
+
+export const LaunchConfig: FC = () => {
+  const dispatch = useDispatch()
+  const allLaunchFiles = useSelector(selectAllLaunchFiles)
+  const [isLoading, setIsLoading] = useState(false)
+  const [connectionState] = useActor(rosService)
+
+  const onClick = useCallback(
+    (fileName: string, packageName: string) => {
+      setIsLoading(true)
+      rosClient
+        .callService(
+          {
+            name: '/launchHandler/launchFile',
+          },
+          { package: packageName, fileName }
+        )
+        .then((res: unknown) => {
+          const messageData = res as LaunchMsg
+          setIsLoading(false)
+          if (messageData.isLaunched) {
+            dispatch(launchFilesSlice.actions.launchFile(messageData.fileName))
+            toast.success(messageData.message)
+          } else {
+            dispatch(launchFilesSlice.actions.killFile(messageData.fileName))
+            toast.error(messageData.message)
+          }
+        })
+        .catch((e: string) => {
+          setIsLoading(false)
+          toast.error(e)
+        })
+    },
+    [dispatch]
+  )
+
+  const onClickLaunchAll = useCallback(() => {
+    allLaunchFiles.forEach((element) => {
+      if (!element.isLaunched) {
+        onClick(element.fileName, element.packageName)
+      }
+    })
+  }, [allLaunchFiles, onClick])
+
+  const onClickKillAll = useCallback(() => {
+    allLaunchFiles.forEach((element) => {
+      if (element.isLaunched) {
+        onClick(element.fileName, element.packageName)
+      }
+    })
+  }, [allLaunchFiles, onClick])
+
+  const onClickRefresh = () => {
+    refreshLaunchedFiles()
+  }
+
+  const refreshLaunchedFiles = useCallback(() => {
+    rosClient
+      .callService({
+        name: '/launchHandler/getAllLaunchedFiles',
+      })
+      .then((res: unknown) => {
+        const launchedFiles = res as LaunchedFiles
+        for (const element of allLaunchFiles) {
+          if (
+            !element.isLaunched &&
+            launchedFiles.fileNames.includes(element.fileName)
+          ) {
+            dispatch(launchFilesSlice.actions.launchFile(element.fileName))
+          } else if (
+            element.isLaunched &&
+            !launchedFiles.fileNames.includes(element.fileName)
+          ) {
+            dispatch(launchFilesSlice.actions.killFile(element.fileName))
+          }
+        }
+      })
+      .catch((e: string) => {
+        toast.error(e)
+      })
+  }, [allLaunchFiles, dispatch])
+
+  useEffect(() => {
+    if (connectionState.matches('connected')) {
+      refreshLaunchedFiles()
+    } else {
+      allLaunchFiles.forEach((element) => {
```

Comment:
C'est un petit détail, mais tu alterne entre utiliser des `for ... of` et `forEach()` pour itérer sur cette liste la. Générallement c'est mieux de garder un style consistent.